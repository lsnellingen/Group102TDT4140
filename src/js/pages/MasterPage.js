// Libs
import { Link } from 'react-router';
import React, { PropTypes } from 'react';
import DocumentTitle from 'react-document-title';
import { LoginLink } from 'react-stormpath';

// Components
import Sidebar from './Sidebar';
import Header from './Header';
import Footer from './Footer';


export default class MasterPage extends React.Component {
  render() {
    return (
      <DocumentTitle title='eduBot - Feedback for students and lecturers'>
        <div className="MasterPage">
          <div className="row">
            <Header />
            <div className= 'col-xs-2'>
              <Sidebar />
            </div>
            <div className = 'col-xs-10'>
              { this.props.children }
            </div>
          </div>
          <Footer />
        </div>
      </DocumentTitle>
    );
  }
}
