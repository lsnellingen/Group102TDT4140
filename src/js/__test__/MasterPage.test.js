import React from 'react';
import ReactDOM from 'react-dom';
import MasterPage from '../pages/MasterPage';
import renderer from 'react-test-renderer';
import { RegistrationForm, LoginLink } from 'react-stormpath';
import {shallow} from 'enzyme';

describe('MasterPage', () => {
  var ReactTestUtils = require('react-addons-test-utils') // ES5 with npm

    test('renders without crashing', () => {


      const tree = shallow(
        <MasterPage/>);


      expect(tree).toMatchSnapshot();




    });
});
