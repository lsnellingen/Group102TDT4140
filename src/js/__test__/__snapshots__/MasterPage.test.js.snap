// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasterPage renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <SideEffect(DocumentTitle)
        title="eduBot - Feedback for students and lecturers"
    >
        <div
            className="MasterPage"
        >
            <div
                className="row"
            >
                <Header />
                <div
                    className="col-sm-2 col-xs-12 marginTop50"
                >
                    <Sidebar />
                </div>
                <div
                    className="col-sm-10 col-xs-12 marginTop50"
                />
            </div>
        </div>
    </SideEffect(DocumentTitle)>
</div>,
  "nodes": Array [
    <div>
      <SideEffect(DocumentTitle)
            title="eduBot - Feedback for students and lecturers"
      >
            <div
                  className="MasterPage"
            >
                  <div
                        className="row"
                  >
                        <Header />
                        <div
                              className="col-sm-2 col-xs-12 marginTop50"
                        >
                              <Sidebar />
                        </div>
                        <div
                              className="col-sm-10 col-xs-12 marginTop50"
                        />
                  </div>
            </div>
      </SideEffect(DocumentTitle)>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <MasterPage />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": MasterPage {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <SideEffect(DocumentTitle)
                    title="eduBot - Feedback for students and lecturers"
          >
                    <div
                              className="MasterPage"
                    >
                              <div
                                        className="row"
                              >
                                        <Header />
                                        <div
                                                  className="col-sm-2 col-xs-12 marginTop50"
                                        >
                                                  <Sidebar />
                                        </div>
                                        <div
                                                  className="col-sm-10 col-xs-12 marginTop50"
                                        />
                              </div>
                    </div>
          </SideEffect(DocumentTitle)>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <SideEffect(DocumentTitle)
                    title="eduBot - Feedback for students and lecturers"
          >
                    <div
                              className="MasterPage"
                    >
                              <div
                                        className="row"
                              >
                                        <Header />
                                        <div
                                                  className="col-sm-2 col-xs-12 marginTop50"
                                        >
                                                  <Sidebar />
                                        </div>
                                        <div
                                                  className="col-sm-10 col-xs-12 marginTop50"
                                        />
                              </div>
                    </div>
          </SideEffect(DocumentTitle)>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MasterPage />,
}
`;
